### PR TITLE
Add Python 3.9 compatibility.

### LINKED ISSUE #423: Importing ABC from collections module was removed in Python 3.9
https://github.com/python-social-auth/social-core/blob/07c18a81327050334ea71576d397f5d3bb8b0757/social_core/utils.py#L114

### PR DESCRIPTION
**Pull request by tirkarthi**
* Import ABC from collections.abc instead of collections for Python 3.9 compatibility.
* Fix warnings regarding invalid escape sequences and base64.

Fixes #423

### CODE REVIEW

File: social_core/utils.py [hunk 1]
```diff
@@ -111,7 +115,7 @@ def sanitize_redirect(hosts, redirect_to):
 
 def user_is_authenticated(user):
     if user and hasattr(user, 'is_authenticated'):
-        if isinstance(user.is_authenticated, collections.Callable):
+        if isinstance(user.is_authenticated, Callable):
```
Comment (serhiy-storchaka) [inline social_core/utils.py]: You can just use built-in function `callable()`.
```suggestion
        if callable(user.is_authenticated):
```

File: social_core/utils.py [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
 import unicodedata
-import collections
 import functools
 import hmac
 import logging
 
+try:
+    from collections.abc import Callable
+except ImportError:
+    from collections import Callable
```
Comment (andersk) [inline social_core/utils.py:12]: Could use `six.moves.collections.abc.Callable` on both versions.

Comment (andersk) [reply social_core/utils.py:12]: Although actually, the `callable()` builtin would do just as well in these cases.

**Comment by andersk** [comment]
Reopened as #473.

**Comment by nijel** [comment]
Already merged as https://github.com/python-social-auth/social-core/pull/473

**Pull request closed without merging**